infra/swapi.go

```diff
@@ -0,0 +1,80 @@
+package infra
+
+import (
+	"fmt"
+	"path/filepath"
+	"sort"
+	"strconv"
+
+	"github.com/SergioSanchezWizeline/academy-go-q12021/domain/model"
+
+	"github.com/go-resty/resty/v2"
+)
+
+type SwapiPeople struct {
+	URL  string `json:"url"`
+	Name string `json:"name"`
+}
+
+type SwapiResult struct {
+	Count   int           `json:"count"`
+	Next    string        `json:"next"`
+	Results []SwapiPeople `json:"results"`
+}
+
+func LoadSwapiCharacters() ([]*model.Character, error) {
+
+	jobs := make(chan string, 10)
+	people := make(chan SwapiPeople, 100)
+
+	fmt.Println("Fetch start")
+
+	jobs <- "https://swapi.dev/api/people/"
+
+	client := resty.New()
+
+	go worker(client, jobs, people)
+	go worker(client, jobs, people)
+
+	characters := []*model.Character{}
+	for p := range people {
+		path := filepath.Dir(p.URL)
+		idStr := filepath.Base(path)
+		id, _ := strconv.Atoi(idStr)
+		characters = append(characters, &model.Character{ID: id, Name: p.Name})
+	}
+
+	sort.Slice(characters, func(i, j int) bool {
+		return characters[i].ID < characters[j].ID
+	})
+	fmt.Println("Fetch complete")
+	return characters, nil
+}
+
+func worker(client *resty.Client, jobs chan string, people chan<- SwapiPeople) {
+
+	for url := range jobs {
+		response, err := client.R().
+			SetHeader("Accept", "application/json").
+			SetResult(SwapiResult{}).
+			Get(url)
+
+		if err != nil {
+			fmt.Printf("error loading swapi people: %v \n", err)
```

Comment:
You can drop the `else` by adding a `continue` after this line.

Reply:
Done